dev/reference/best-practices/pd-scheduling.md

```diff
@@ -0,0 +1,269 @@
+    - The restrictions of cluster topology. For example, in a cluster of 3 replicas and 3 data centers, 3 replicas of each Region are distributed in different data centers due to replica isolation. If the number of Stores of these data centers are different, the final scheduling reaches a balanced but globally unbalanced state in each data center.
+
+### The speed of taking nodes offline is slow
+
+This scenario requires examining the generation and execution of Operators through related metrics.
+
+When an Operator is successfully generated but processes slow, possible reasons are:
+
+- The schedule speed is limited by default. You can adjust `leader-schedule-limit` or `replica-schedule-limit` to a larger value. Similarly, `max-pending-peer-count` and `max-snapshot-count` can also be properly enlarged.
+- Other scheduling tasks are running concurrently and competing in the system. You can refer to the solution in [the previous section](#leaderregion-is-not-evenly-distributed).
+- When you take a single node offline, a number of Region Leaders to be operated are concentrated on the offline node (about 1/3 under the configuration of 3 replicas), so the speed is limited by the speed at which this single node generates Snapshots. You can speed it up by manually adding an `evict-leader-scheduler` to migrate Leaders.
+
+If the corresponding Operator fails to generate, possible reasons are:
+
+- The Operator is stopped, or `replica-schedule-limit` is set to 0.
+- There is no proper node to migrate Regions. For example, if the capacity of nodes that replace the nodes of same Label is larger than 80%, PD will stop scheduling to avoid running out of storage space. In such case, you need to add more nodes or delete some data to free space.
+
+### The speed of putting nodes online is slow
+
+Currently, to take nodes online is scheduled through balance region mechanism, so you can refer to [Leader/Region is not evenly distributed](#leaderregion-is-not-evenly-distributed) for troubleshooting.
+
+### Hot Regions are not evenly distributed
+
+Hot Regions scheduling generally has the following problems:
+
+- There is a majority of hot Regions, but the scheduling speed cannot keep up with them to redistribute hot Regions in time.
+
+    **Solution**: adjust `hot-region-schedule-limit` to a larger value, and reduce the limit quota of other schedulers to speed up hot Regions scheduling. Or you can adjust `hot-region-cache-hits-threshold` to a smaller value to make PD sensitive to traffic changes.
+
+- A single Region with extensive traffics. For example, to scan a small table extensively is required in the production environment, which can also be detected from PD metrics. Since a single hotspot cannot be resolved by redistributing, you need to manually add a `split-region` Operator to redistribute such a Region.
```

Comment:
```suggestion
- Hotspot formed on a single Region, for example, a small table intensively scanned by massive amount of requests。 This can also be detected from PD metrics. Since you cannot actually scatter a single hotspot, you need to manually add a `split-region` Operator to split such a Region.
```